src/map/intif.c

```diff
@@ -439,6 +439,140 @@ int intif_request_registry(struct map_session_data *sd, int flag)
 	return 0;
 }
 
+//=================================================================
+// Account Storage
+//-----------------------------------------------------------------
+
+/**
+ * Request the inter-server for a character's storage data.
+ * @packet 0x3010  [out] <account_id>.L
+ * @param  sd      [in]  pointer to session data.
+ */
+void intif_request_account_storage(const struct map_session_data *sd)
+{
+	nullpo_retv(sd);
+
+	/* Check for character server availability */
+	if (intif->CheckForCharServer())
+		return;
+
+	WFIFOHEAD(inter_fd, 6);
+	WFIFOW(inter_fd, 0) = 0x3010;
+	WFIFOL(inter_fd, 2) = sd->status.account_id;
+	WFIFOSET(inter_fd, 6);
+}
+
+/**
+ * Parse the reception of account storage from the inter-server.
+ * @packet 0x3805 [in] <packet_len>.W <account_id>.L <struct item[]>.P
+ * @param  fd     [in] file/socket descriptor.
+ */
+void intif_parse_account_storage(int fd)
+{
+	int account_id = 0, payload_size = 0, storage_count = 0;
+	int i = 0;
+	struct map_session_data *sd = NULL;
+
+	Assert_retv(fd > 0);
+
+	payload_size = RFIFOW(fd, 2) - 8;
+
+	if ((account_id = RFIFOL(fd, 4)) == 0 || (sd = map->id2sd(account_id)) == NULL) {
+		ShowError("intif_parse_account_storage: Session pointer was null for account id %d!\n", account_id);
+		return;
+	}
+
+	if (sd->storage.received == true) {
+		ShowError("intif_parse_account_storage: Multiple calls from the inter-server received.\n");
+		return;
+	}
+
+	storage_count = (payload_size/sizeof(struct item));
+
+	VECTOR_ENSURE(sd->storage.item, storage_count, 1);
+
+	sd->storage.aggregate = storage_count; //< Total items in storage.
+
+	for (i = 0; i < storage_count; i++) {
+		const struct item *it = RFIFOP(fd, 8 + i * sizeof(struct item));
+		VECTOR_PUSH(sd->storage.item, *it);
+	}
+
+	sd->storage.received = true; //< Mark the storage state as received.
+	sd->storage.save = false; //< Initialize the save flag as false.
+}
```

Comment:
This seems a good place to re-attempt a second itemcheck, in case the storage wasn't available yet at the time the first itemcheck runs (this time it'll only run the storage part, since the other flags have been unset already)